mmf/datasets/builders/vqacp_v2/dataset.py

```diff
@@ -0,0 +1,66 @@
+# Copyright (c) Facebook, Inc. and its affiliates.
+from typing import Type, Union
+
+import torch
+
+from mmf.common.sample import Sample
+from mmf.common.typings import MMFDatasetConfigType
+from mmf.datasets.builders.vqacp_v2.database import (
+    VQACPv2AnnotationDatabase,
+    VQACPv2ImageDatabase,
+)
+from mmf.datasets.mmf_dataset import MMFDataset
+
+
+class VQACPv2Dataset(MMFDataset):
```

Comment:
If the code is mostly same with respect to OKVQADataset, some refactoring or inheriting would reduce further duplication.

Reply:
It is mostly the same, except for the get_image_path function. I'm not sure how to inherit/refactor it without making some changes to the OKVQADataset. Do you have any suggestions for this?

Reply:
Instead of extending from the `MMFDataset` extend from the `OKVQADataset` and override the necessary functions.